src/strpkg/callclusters.nim

```diff
@@ -0,0 +1,70 @@
+import algorithm
+import hts/bam
+import strformat
+
+import ./cluster
+import ./collect
+import ./utils
+import ./genotyper
+import ./genome_strs
+import ./extract
+
+# returns reads assign to locus and removes them from table.
+# Updates Bounds with number of supporting reads
+proc assign_reads_locus*(locus: var Bounds, treads_by_tid_rep: TableRef[tid_rep, seq[tread]]): seq[tread] =
+    var key: tid_rep = (locus.tid, locus.repeat.as_array)
+    var trs = treads_by_tid_rep.getOrDefault(key, @[])
+
+    var left_most = if locus.left_most == 0: 0'u32 else: locus.left_most - 1'u32
+    var li = lowerBound(trs, tread(position: left_most), proc(a, b:tread):int =
+      return cmp(a.position, b.position)
+    )
+    var ri = upperBound(trs, tread(position: locus.right_most), proc(a, b:tread):int =
+      return cmp(a.position, b.position)
+    )
+    when defined(debug):
+      if ri - li > 0:
+        stderr.write_line &"[strling] got {ri - li} treads for locus: {locus} with indexes {li}..{ri}"
+
+    if trs.len > 0:
+      # now we have a the subset of treads that support the given locus
+      result = trs[li..<ri]
+
+      # remove these from the table
+      treads_by_tid_rep[key] = trs[0..<li]
+      if ri < trs.high:
+        treads_by_tid_rep[key].add(trs[ri + 1..trs.high])
+
+    # Force reporting of loci from input files
+    # even if they have insufficient supporting reads
+    locus.force_report = true 
+    # Update read counts in Bounds
+    locus.n_total = 0
+    locus.n_left = 0
+    locus.n_right = 0
+    for r in result:
+      locus.n_total.inc
+      if r.split == Soft.right:
+        locus.n_right.inc
+      elif r.split == Soft.left:
+        locus.n_left.inc
+
+proc check_cluster*(c: Cluster, min_clip: uint16, min_clip_total: uint16): (Bounds, bool) =
```

Comment:
`check_cluster` is kind of a mis-nomer. could just be `bounds` since it has a different signature than the other `cluster.bounds`.

Reply:
Good point. I think it started as a cluster checker the evolved to make bounds and I forgot to change it. Fixed.